util/src/main/java/io/grpc/util/AdvancedTlsX509TrustManager.java

```diff
@@ -43,30 +44,38 @@
 /**
  * AdvancedTlsX509TrustManager is an {@code X509ExtendedTrustManager} that allows users to configure
  * advanced TLS features, such as root certificate reloading, peer cert custom verification, etc.
+ * We expect only one of {@link AdvancedTlsX509TrustManager#useSystemDefaultTrustCerts()},
```

Comment:
Can you elaborate on what happens if the user does not respect this?

Reply:
Theoretically speaking, a user can do the following trick:

1. Some process (non gRPC) updates the cert file
2. User spawns a thread to periodically call updateTrustCredentialsFromFile(file)
3. User calls updateTrustCredentialsFromFile(File trustCertFile, long period, TimeUnit unit,
      ScheduledExecutorService executor) multiple times without cancelling the result

As a result an outdated version of creds might be used till the next update. It's not truly a synchronization issue (that's why we don't guard this methods with locks) but a similar to it - so we decided to add a warning. Definitely the example is a made up one and we don't expect a user do anything like that

Reply:
Makes sense, thanks for clarifying.